python/pyfory/serializer.py

```diff
@@ -703,6 +703,59 @@ def to_buffer(self) -> "Buffer":
         return Buffer(self.binary)
 
 
+class StatefulSerializer(CrossLanguageCompatibleSerializer):
+    """
+    Serializer for objects that support __getstate__ and __setstate__.
+    Uses Fory's native serialization for better cross-language support.
+    """
+
+    def __init__(self, fory, cls):
+        super().__init__(fory, cls)
+        self.cls = cls
+
+    def write(self, buffer, value):
+        # Get the state using __getstate__
+        state = value.__getstate__()
+
+        # Check for constructor arguments
+        args = ()
+        kwargs = {}
+
+        if hasattr(value, "__getnewargs_ex__"):
```

Comment:
Do we need to cache this method ref as a field in this serializer? It's rare that users override this mehod on a instance variable.

Reply:
Or we can left this in future PR, current implementation has better compatibility

Reply:
Looks good